shm/island.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include "coordinates.hpp"
+
+class Island {
+public:
+    Island(const Coordinates& position)
```

Comment:
The definition should be in .cpp files. Only simple getter/setters are allowed in .hpp